pubsub/fakepubsub/fake_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fakepubsub
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cloud/pubsub/driver"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestSendReceive(t *testing.T) {
+	ctx := context.Background()
+	top := CreateTopic()
+	sub := CreateSubscription(top, 3*time.Second)
+	want := []*driver.Message{
+		{Body: []byte("a")},
+		{Body: []byte("b")},
+		{Body: []byte("c")},
+	}
+	if err := top.SendBatch(ctx, want); err != nil {
+		t.Fatal(err)
+	}
+	got, err := sub.ReceiveBatch(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	less := func(x, y *driver.Message) bool { return bytes.Compare(x.Body, y.Body) < 0 }
+	if diff := cmp.Diff(got, want, cmpopts.SortSlices(less)); diff != "" {
+		t.Error(diff)
+	}
+}
+
+func TestReceive(t *testing.T) {
```

Comment:
OK to do this in a separate PR, but these should be implemented as conformance tests similar to `blob` and `runtimevar`, not tests specific to this implementation.

Reply:
Will leave in for reference, and fix when I write conformance tests.

Reply:
OK. @ijt please note -- if you're working on another driver impl, you shouldn't have to write tests for it (at least not for functionality exposed through the concrete type, which is hopefully most of it).